src/callbacks.c

```diff
@@ -157,6 +157,27 @@ void on_close_all1_activate(GtkMenuItem *menuitem, gpointer user_data)
 }
 
 
+void on_close_allafter1_activate(GtkMenuItem *menuitem, gpointer user_data)
+{
+	guint i;
+	GeanyDocument *cur_doc = user_data;
+
+	if (cur_doc == NULL)
+		cur_doc = document_get_current();
+
+	for (i = cur_doc->index + 1; i < documents_array->len; i++)
+	{
+		GeanyDocument *doc = documents[i];
+
+		if (! doc->is_valid)
```

Comment:
That's a very valid remark. No, the documents_array is not in "tab order".

See document_account_for_unsaved() (but I think it would be better to create a copy and sort it with g_ptr_array_sort() and document_compare_by_tab_order(), it's kind of ugly to loop through the notebook pages directly IMO)